### ISSUE TITLE
Unable to remove / update `extra-scidb-libs` 19.3-3 / 19.11-x

### ISSUE THREAD

**Issue by ksens** [Paradigm4/extra-scidb-libs#26]
Trying to remove `extra-scidb-libs`

```
sudo yum remove extra-scidb-libs-19.3
```

Getting the error:

```sh
Installed size: 8.1 M
Is this ok [y/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
/var/tmp/rpm-tmp.fbMJsg: line 1: /shim/before-remove.sh: No such file or directory
error: %preun(extra-scidb-libs-19.3-3-1.x86_64) scriptlet failed, exit status 127
Error in PREUN scriptlet in rpm package extra-scidb-libs-19.3-3-1.x86_64
  Verifying  : extra-scidb-libs-19.3-3-1.x86_64                                                                                        1/1

Failed:
  extra-scidb-libs-19.3.x86_64 0:3-1

Complete!
```

@martin-resnick maybe you can help with this?

**Comment by rvernica** [comment]
It looks like this is due to a recent change, see this [commit](https://github.com/Paradigm4/extra-scidb-libs/commit/581f2612a71de912c27c3e8a6e9a2d86d59e95ec#diff-ceea2726b51efcf6790390e7153bf894R101) That `before-remove.sh` script is in the Shim [repo](https://github.com/Paradigm4/shim/tree/master/init.d) so I'm not sure what is going on. Maybe a typo in the path...

**Comment by rvernica** [comment]
Is this still an issue?

**Comment by ksens** [comment]
Yes, this is still an issue. @martin-resnick Can you please take a look?

I ran into this again when trying to install `extra-scidb-libs-19.11` on a system where I had just upgraded scidb from 19.3 to 19.11, and was trying to upgrade extra-scidb-libs from 19.3 to 19.11.

**Comment by ksens** [comment]
Haven't run into this recently. Closing issue.

**Comment by ksens** [comment]
Reopening issue (pasting from report at #30). This problem still remains.

@JAKinchen @rvernica Could one of you take a look at this? 

==========

On a customer site, I had `extra-scidb-libs-19.11-1`. Tried updating:

```sh
sudo yum update extra-scidb-libs-19.11
```

... it gave the following log:
```
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
extra-scidb-libs-19.11-2-1.x86_64.rpm                                                                                                                                                              | 2.0 MB  00:00:02
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : extra-scidb-libs-19.11-2-1.x86_64                                                                                                                                                                      1/2
/var/tmp/rpm-tmp.jWYPdn: line 1: /shim/before-remove.sh: No such file or directory
error: %preun(extra-scidb-libs-19.11-1-1.x86_64) scriptlet failed, exit status 127
Error in PREUN scriptlet in rpm package extra-scidb-libs-19.11-1-1.x86_64
  Verifying  : extra-scidb-libs-19.11-2-1.x86_64                                                                                                                                                                      1/2
  Verifying  : extra-scidb-libs-19.11-1-1.x86_64                                                                                                                                                                      2/2

Updated:
  extra-scidb-libs-19.11.x86_64 0:2-1

Failed:
  extra-scidb-libs-19.11.x86_64 0:1-1

Complete!
```

Now the system had two versions installed:

```sh
yum list installed | grep extra-scidb
# extra-scidb-libs-19.11.x86_64         1-1                        @scidb-extra
# extra-scidb-libs-19.11.x86_64         2-1                        @scidb-extra
```

If I try to remove the old version, I get an error:

```sh
sudo yum remove extra-scidb-libs-19.11-1
```

gives:

```
Running transaction
/var/tmp/rpm-tmp.ZOCBtX: line 1: /shim/before-remove.sh: No such file or directory
error: %preun(extra-scidb-libs-19.11-1-1.x86_64) scriptlet failed, exit status 127
Error in PREUN scriptlet in rpm package extra-scidb-libs-19.11-1-1.x86_64
  Verifying  : extra-scidb-libs-19.11-1-1.x86_64                                                                                                                                                                      1/1

Failed:
  extra-scidb-libs-19.11.x86_64 0:1-1

Complete!
```

**Comment by rvernica** [comment]
I tried to reproduce this on a clean CentOS 7 `19.11` container but it seems to work fine:
```
# yum update extra-scidb-libs-19.11
...
Resolving Dependencies
--> Running transaction check
---> Package extra-scidb-libs-19.11.x86_64 0:1-1 will be updated
---> Package extra-scidb-libs-19.11.x86_64 0:2-1 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

========================================================================
 Package                    Arch       Version    Repository       Size
========================================================================
Updating:
 extra-scidb-libs-19.11     x86_64     2-1        scidb-extra     2.0 M

Transaction Summary
========================================================================
Upgrade  1 Package

Total download size: 2.0 M
Is this ok [y/d/N]: y
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
extra-scidb-libs-19.11-2-1.x86_64.rpm              | 2.0 MB   00:01     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : extra-scidb-libs-19.11-2-1.x86_64                    1/2 
  Cleanup    : extra-scidb-libs-19.11-1-1.x86_64                    2/2 
  Verifying  : extra-scidb-libs-19.11-2-1.x86_64                    1/2 
  Verifying  : extra-scidb-libs-19.11-1-1.x86_64                    2/2 

Updated:
  extra-scidb-libs-19.11.x86_64 0:2-1                                   

Complete!
```

**Comment by rvernica** [comment]
There is one `before-remove.sh` located here:
```
# find / -name before-remove.sh
/opt/scidb/19.11/shim/before-remove.sh
```
I manually removed it before the update. Then I ran the `yum update` and it works just fine and puts the file back.
```
# ll /opt/scidb/19.11/shim/before-remove.sh
-rwxr-xr-x. 1 root root 704 Apr 14 14:05 /opt/scidb/19.11/shim/before-remove.sh

# cat /opt/scidb/19.11/shim/before-remove.sh
#!/bin/bash
# shim
# package pre-uninstallation script
# This script is run by the package management system before package
# is uninstalled.

if test -n "$(which systemctl 2>/dev/null)"; then
# SystemD
  systemctl -q stop shimvc 2>/dev/null || true
  systemctl -q disable shimvc 2>/dev/null || true
  rm -f /lib/systemd/system/shim.service
  systemctl -q daemon-reload 2>/dev/null || true
elif test -n "$(which update-rc.d 2>/dev/null)"; then
# Ubuntu
  service shimsvc stop
  update-rc.d -f shimsvc remove
  rm -f /etc/init.d/shimsvc
elif test -n "$(which chkconfig 2>/dev/null)"; then
# RHEL sysV
  service shimsvc stop
  chkconfig --del shimsvc
  chkconfig shimsvc off
  rm -f /etc/init.d/shimsvc
fi
```
A few notes for you:
- Is it a CentOS or RedHat instance? Which version?
- Can you check that the `before-remove.sh` script is there? Verify it has the same permissions as above.
- See if you can (re-)create the file. I pasted the contents of the one I have above. The original version is [here](https://github.com/Paradigm4/shim/blob/v19.11/init.d/before-remove.sh)
- Can you force-remove the package? Maybe you got stuck in an inconsistent state. There were a few significant changes to the package in the last version of the `19.3` branch for Shim.

**Comment by ksens** [comment]
Thanks @rvernica 

Will respond in detail by Thursday.

**Comment by rvernica** [comment]
We investigated this on the affected system. It appears that this error line is the key:
```
/var/tmp/rpm-tmp.ZOCBtX: line 1: /shim/before-remove.sh: No such file or directory
```
The error is the result of [this line](https://github.com/Paradigm4/extra-scidb-libs/blob/master/specs/extra-scidb-libs.spec#L101). Essentially the `$SCIDB_INSTALL_PATH` environment variable is not set. Setting this variable is discussed in the SciDB install [documentation](https://paradigm4.atlassian.net/wiki/spaces/scidb/pages/729876447/Installing+SciDB+Community+Edition#InstallingSciDBCommunityEdition-ConfiguringEnvironmentVariables) resolved the problem.

**Comment by rvernica** [comment]
In dad63b859e1bf2e3b7b0c967ac1a0336daa8c0dc we added an additional step to the Travis tests to also remove the package.

**Comment by ksens** [comment]
One of my setups at customer A  / revealgenomics was giving unusual problems with `shimsvc` somehow getting removed from the system. Not sure how that happened -- will need to investigate. 

For now, I resolved the issue this by removing `extra-scidb-libs` and reinstalling it. Storing my commands here for future reference:

```sh
# as root
export SCIDB_INSTALL_PATH=/opt/scidb/19.11
yum remove extra-scidb-libs-19.11
yum --enablerepo=paradigm4-19p11-extra install extra-scidb-libs-19.11
```

**Comment by rvernica** [comment]
I think this is the updated link https://paradigm4.atlassian.net/wiki/spaces/SD911/pages/1973387527/Installing+SciDB+Community+Edition#InstallingSciDBCommunityEdition-ConfiguringEnvironmentVariables

**Comment by rvernica** [comment]
In case the environment variable is not set, we added a default value in the uninstall steps to take care of setting this variable. Maybe you got an old version of the package or the default was not correct in your setup.

[Here](https://github.com/Paradigm4/extra-scidb-libs/blob/c26d2b8e121d862dc5971563ca1832bdc3864522/specs/extra-scidb-libs.spec#L155) is how we set this variable in case it is not set. This change is both in `19.11` and `21.8` versions of the package.
